src/lib/pubkey/ec_group/ec_group.cpp

```diff
@@ -266,14 +266,46 @@ std::pair<std::shared_ptr<EC_Group_Data>, bool> EC_Group::BER_decode_EC_group(st
          throw Decoding_Error("Invalid ECC cofactor parameter");
       }
 
-      const auto [g_x, g_y] = Botan::OS2ECP(base_pt.data(), base_pt.size(), p, a, b);
+      const size_t p_bytes = p.bytes();
+      if(base_pt.size() != 1 + p_bytes && base_pt.size() != 1 + 2 * p_bytes) {
+         throw Decoding_Error("Invalid ECC base point encoding");
+      }
+
+      auto [g_x, g_y] = [&]() {
+         const uint8_t hdr = base_pt[0];
+
+         if(hdr == 0x04 && base_pt.size() == 1 + 2 * p_bytes) {
+            BigInt x = BigInt::decode(&base_pt[1], p_bytes);
+            BigInt y = BigInt::decode(&base_pt[p_bytes + 1], p_bytes);
+
+            if(x < p && y < p) {
+               return std::make_pair(x, y);
```

Comment:
Performance nit: I'm quite sure, for optimal performance you'd have to `std::move()` x and y into the result pair (also in the else-if branch). Alternatively, you could pre-initialize a result pair in the beginning and decode the `BigInt`s straight into `.first` and `.second`. Then RVO should do the trick.

Reply:
Fair enough but considering what else is happening here - at best just testing the curve equation and possibly a point decompression, a couple of extra copies seems negligible.